cpu/cortexm_common/thread_arch.c

```diff
@@ -286,9 +286,17 @@ void __attribute__((naked)) __attribute__((used)) isr_pendsv(void) {
     /* save context by pushing unsaved registers to the stack */
     /* {r0-r3,r12,LR,PC,xPSR,s0-s15,FPSCR} are saved automatically on exception entry */
     ".thumb_func                      \n"
+
+    /* skip context saving if sched_active_thread == NULL */
+    "ldr    r1, =sched_active_thread  \n" /* r1 = &sched_active_thread  */
+    "ldr    r1, [r1]                  \n" /* r1 = sched_active_thread   */
+    "cmp    r1, #0                    \n" /* if r1 == NULL:             */
+    "beq context_restore              \n" /*     goto context_restore   */
```

Comment:
Skipping context switching depending on `sched_active_thread` should go into its own commit.

Reply:
ok, will re-shuffle when squashing